packages/api/src/command/medical/patient/converage-assessment-get.ts

```diff
@@ -0,0 +1,143 @@
+  downloadStatus: string | undefined;
+  docCount: number | undefined;
+  convertStatus: string | undefined;
+  fhirCount: number;
+  fhirDetails: string;
+  mrSummaryUrl: string | undefined;
+};
+
+export type PatientWithCoverageAssessment = Patient &
+  Partial<Omit<CoverageAssessment, "patientId">>;
+
+export async function getCoverageAssessments({
+  cxId,
+  patients,
+}: {
+  cxId: string;
+  patients: Patient[];
+}): Promise<PatientWithCoverageAssessment[]> {
+  const { log } = out(`getCoverageAssessments - cxId ${cxId}`);
+  const patientsWithAssessment: PatientWithCoverageAssessment[] = [];
+
+  async function getCoverageAssessmentWrapper({
+    cxId,
+    patient,
+    patientsWithAssessment,
+    log,
+  }: {
+    cxId: string;
+    patient: Patient;
+    patientsWithAssessment: PatientWithCoverageAssessment[];
+    log: typeof console.log;
+  }): Promise<void> {
+    try {
+      const coverageAssessment = await getCoverageAssessment({ cxId, patient, log });
+      patientsWithAssessment.push({
+        ...patient,
+        ...coverageAssessment,
+      });
+    } catch (error) {
+      const msg = `Failed to get coverage assessment. Cause: ${errorToString(error)}`;
+      capture.error(msg, {
+        extra: {
+          cxId,
+          patientId: patient.id,
+          context: "coverage-assessment.get",
+        },
+      });
+      patientsWithAssessment.push(patient);
+    }
+  }
+
+  await executeAsynchronously(
+    patients.map(patient => {
+      return { cxId, patient, patientsWithAssessment, log };
+    }),
+    getCoverageAssessmentWrapper,
+    { numberOfParallelExecutions: 50 }
+  );
+
+  return patientsWithAssessment;
+}
+
+async function getCoverageAssessment({
+  cxId,
+  patient,
+  log,
+}: {
+  cxId: string;
+  patient: Patient;
+  log: typeof console.log;
+}): Promise<CoverageAssessment> {
+  const mrSummaryFileName = createMRSummaryFileName(cxId, patient.id, "json");
+  const [fhirResources, mrSummaryUrl] = await Promise.all([
+    countResources({ patient }),
+    getMrSummaryUrl(mrSummaryFileName, log),
+  ]);
+
+  const download = patient.data.documentQueryProgress?.download;
+  const convert = patient.data.documentQueryProgress?.convert;
+  const downloadStatus = download?.status;
+  const docCount = download?.successful;
+  const convertStatus = convert?.status;
+  const fhirCount = fhirResources.total;
+  const fhirDetails = JSON.stringify(fhirResources.resources).replaceAll(",", " ");
+
+  return {
+    patientId: patient.id,
+    downloadStatus,
+    docCount,
+    convertStatus,
+    fhirCount,
+    fhirDetails,
+    mrSummaryUrl,
+  };
+}
+
+async function getMrSummaryUrl(
+  fileName: string,
+  log: typeof console.log
+): Promise<string | undefined> {
+  const s3Utils = getS3UtilsInstance();
+  try {
+    const object = await s3Utils.getFileInfoFromS3(fileName, bucket);
+    if (object.exists) {
+      return await s3Utils.getSignedUrl({
+        bucketName: bucket,
+        fileName,
+      });
```

Comment:
The default duration is 3min, shouldn't we generate it with maybe 10-15min since we might have a lot of files to work with on the OpsDash?

Reply:
Made it an hour -- didn't realize there's a default timeout.